### PR TITLE
server: fix stale regions info with overlaps that should be removed

### PR DESCRIPTION
**Pull request by Connor1996**
when having detected overlapped regions, we just remove them from tree. We should remove them from regions, or pd-ctl's region command still show the regions with overlapped key range.

### CODE REVIEW

**Comment by disksing** [comment]
LGTM. Don't forget to fix CI.

File: server/cache_test.go [hunk 1]
```diff
@@ -625,7 +626,11 @@ func mustSaveStores(c *C, kv *core.KV, n int) []*metapb.Store {
 func mustSaveRegions(c *C, kv *core.KV, n int) []*metapb.Region {
 	regions := make([]*metapb.Region, 0, n)
 	for i := 0; i < n; i++ {
-		region := &metapb.Region{Id: uint64(i)}
+		region := &metapb.Region{
+			Id:       uint64(i),
+			StartKey: []byte(fmt.Sprintf("%d", i)),
+			EndKey:   []byte(fmt.Sprintf("%d", i+1)),
```
Comment (disksing) [inline server/cache_test.go]: Consider i is 9, EndKey will be less than StartKey. Better to use '%20d' instead of '%d'.

Comment (siddontang) [reply server/cache_test.go]: why not using bigendian format directly?

Comment (siddontang) [reply server/cache_test.go]: seem you should abstract a function to initialize the region.

Comment (disksing) [reply server/cache_test.go]: Binary format is hard to read.

**Comment by disksing** [comment]
LGTM. PTAL @siddontang

File: server/cache_test.go [hunk 2]
```diff
@@ -625,7 +626,11 @@ func mustSaveStores(c *C, kv *core.KV, n int) []*metapb.Store {
 func mustSaveRegions(c *C, kv *core.KV, n int) []*metapb.Region {
 	regions := make([]*metapb.Region, 0, n)
 	for i := 0; i < n; i++ {
-		region := &metapb.Region{Id: uint64(i)}
+		region := &metapb.Region{
```
Comment (siddontang) [inline server/cache_test.go]: please abstract these to a function

**Comment by siddontang** [comment]
any test to cover this?

**Comment by siddontang** [comment]
LGTM

**Pull request merged**